### PR TITLE
Embedded Kube Cluster

### PR DESCRIPTION
**Pull request by tbak**
+ integration tests migration off Fenzo/Mesos to Kubernetes (embedded)

### CODE REVIEW

File: titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/scenario/ScenarioTemplates.java [hunk 1]
```diff
@@ -57,46 +62,42 @@
...
+                return jobScenarioBuilder
+                        .schedule()
+                        .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.Launched))
+                        .allTasks(taskScenarioBuilder -> taskScenarioBuilder.transitionTo(TaskState.StartInitiated, TaskState.Started))
+                        .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.StartInitiated, TaskState.Started));
+            }
+            return jobScenarioBuilder
+                    .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdateSkipOther(TaskState.Launched))
+                    .allTasks(taskScenarioBuilder -> taskScenarioBuilder.transitionTo(TaskState.StartInitiated, TaskState.Started))
+                    .allTasks(taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdates(TaskState.StartInitiated, TaskState.Started));
```
Comment (fabiokung) [inline titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/scenario/ScenarioTemplates.java:85]: do we still need this code path? Shouldn't everything be `isKube() => true`?

Comment (fabiokung) [reply titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/scenario/ScenarioTemplates.java:85]: if we do need this, perhaps line 68 also needs a `isKube()` check?

File: titus-testkit/src/main/java/com/netflix/titus/testkit/embedded/cell/master/EmbeddedTitusMaster.java [hunk 2]
```diff
@@ -200,6 +213,18 @@ public EmbeddedTitusMaster boot() {
         Stopwatch timer = Stopwatch.createStarted();
         logger.info("Starting Titus Master");
 
+        TitusMasterModule.Mode mode = embeddedKubeCluster != null ? TitusMasterModule.Mode.EMBEDDED_KUBE : TitusMasterModule.Mode.MESOS;
```
Comment (fabiokung) [inline titus-testkit/src/main/java/com/netflix/titus/testkit/embedded/cell/master/EmbeddedTitusMaster.java:216]: Similar question from before, do we still need to maintain two modes?

**Review by fabiokung — APPROVED** [review]

File: titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/scenario/ScenarioTemplates.java (hunk 1 above)
Comment (tbak) [reply titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/scenario/ScenarioTemplates.java:85]: There is a lot of cleanup required to remove Fenzo code branches (+ Mesos CloudSimulator). I am going to do it in the next iteration after merging this PR.

**Pull request merged**